torch_xla/csrc/helpers.cpp

```diff
@@ -527,11 +526,69 @@ std::pair<xla::XlaOp, xla::XlaOp> XlaHelpers::PromoteSecondValue(
 
 xla::Shape XlaHelpers::GetPromotedShape(const xla::Shape& shape1,
                                         const xla::Shape& shape2) {
-  return xla::ShapeUtil::MakeShape(
-      shape1.element_type(),
-      torch::lazy::GetPromotedShape(
-          runtime::util::ToVector<int64_t>(shape1.dimensions()),
-          runtime::util::ToVector<int64_t>(shape2.dimensions())));
+  std::vector<int64_t> dimensions;
+  std::vector<bool> dynamic_dimensions;
+
+  // If the rank of a shape is bigger than then other, fill up the first
+  // dimensions with the ones of the bigger.
+  // Example:
+  //   shape1 = [9, 7, 6, 5, 2]
+  //   shape2 =       [6, 1, 2]
+  // Insert [9, 7] into the dimensions vector.
+  if (shape1.dimensions().size() > shape2.dimensions().size()) {
```

Comment:
I.e. this would be more readable with:

```
if (stape1.simensions.size() > shape2.dimensions().size())
  populateDimensionsFromLhs(shape1, shape2, dimensions, dynamic_dimensions);
else
  populateDimensionsFromLhs(shape2, shape1, dimensions, dynamic_dimensions);
```

The `populateDimensionsFromLhs` wouldn't need control flow either.

Reply:
done